enhancements/platform-validation.md

```diff
@@ -0,0 +1,195 @@
+
+## Summary
+
+A quick start guide to validate Kepler support on specific hardware platform.
+
+## Motivation
+
+Currently, Kepler's test cases focus on code logic verification, lack of actual platform validation.
+
+Kepler's [data](https://github.com/sustainable-computing-io/kepler/tree/main/data) files majorly contain static information, or lack of data sync mechanism.
+
+Test cases and test report are not available, which could be the evidence or check criteria to below concerns:
+
+1. Whether the specific CPU/GPU/FPGA/Accelerator model is well supported in Kepler?
+
+2. Whether the data/metrics of specific power source are correctly exported?
+
+3. Whether the power measurement accuracy is accpetable?
+
+4. More...
+
+### Goals
+
+1. Define workflow to validate specific hardward platform support in Kepler.
+
+2. Design test cases for such validation. (Platform agnostic and specific cases)
+
+3. Define test report format and release process. (General test items result and hardware specific items)
+
+### Non-Goals
+
+1. Platforms which are not supported by Kepler yet should be out of this enhancement's scope.
+
+2. Platforms which do not support power measurement could be low priority for this enhancement, since they depends on model train and the power consumption check criteria is TBD.
+
+## Proposal
+
+We would suggest to add platform validation test cases and define the template of the test report.
+
+The specific tasks could be broken down as follows:
+
+1. Platform validation guide document.
+
+2. Platform validation test cases design and implementation.(Both platform agnostic and specific cases)
+
+3. Platform validation test report format definition, report generation and release process definition.
+
+
+### Workflow Description
+
+Step 1: Create a new local Kubernetes cluster on the test node with target hardware platform.
+
+Bring the cluster up with following [repo](https://github.com/sustainable-computing-io/local-dev-cluster) or just `make cluster-up`
+
+Step 2: Deploy Kepler in the cluster.
+
+There are three kinds of deployment available in Kepler:
+
+Deploy using Helm Chart. Follow guide [here](https://github.com/sustainable-computing-io/kepler-helm-chart/blob/main/README.md).
+
+Deploy using Kepler Operator. Follow guide [here](https://github.com/sustainable-computing-io/kepler-operator/blob/v1alpha1/README.md).
+
+Deploy from Kepler Source Code. Follow guide [here](https://sustainable-computing.io/installation/kepler/).
+
+All the above deployment scenarios related validation should be automated. i.e. locate at $Kepler_ROOT/hack/platform-validation-deploy-from-xxx.sh.
+
+Step 3: Run test cases and get test result.
+
+Step 4: Generate test report and save to somewhere.
+
+Step 5: Delete the local cluster and recover the orignal environment on the test node.
+
+### Implementation Details/Notes/Constraints
+
+Our validation test is local-dev based, in another word, [Kind](https://kind.sigs.k8s.io/) based. 
```

Comment:
as a lot of hardware related things here, kind might not be (at least not today)  should we consider move this from kind to other e.g k3s or other solutions that eligible to do so?

Reply:
I am open to your suggestions:)
Could you identify the specific gap why kind does not support specific hardware platform?

Reply:
at least I knew `kind` has some issue in getting some data accurately :) 
that's the reason why our test/demo rely on baremetal machines and I was told to avoid use such for real calculation
maybe I am wrong after a couple of enhancement since I was first told couple of month ago

Reply:
Thanks for your info, Ji.
Let me do some further investigation and get you tuned.

Reply:
We can expose what we need from the host to Kind. 
For example, when generating a Kind cluster using scripts, the host's `/proc` folder is mounted to facilitate Kepler in visualizing all processes.

The only downside of Kind is its lack of internet connectivity within the cluster. Consequently, we need to upload all the required images to the cluster in order to use them.

Reply:
ok, but seems kind and VM can't get Hardware counter etc info IIRC? 
with that , the test will lack of important part ?

Reply:
My Kind cluster deployment can collect Hardware counters.

Not sure what is the problem that you are mentioning....Can you provide more details?
How your Kind cluster was created?

Reply:
um... ok, I think I remember wrong thing, the kind is not functional on VM ,but it may running on baremetal
https://github.com/sustainable-computing-io/kepler/issues/373#issuecomment-1308512108

so kind is not the problem , it's the VM that has some issue ..

Reply:
guys, sometimes hw counter is not the must, even for VM, we can see that Red Hat guys are trying to submit [patch](https://www.mail-archive.com/qemu-devel@nongnu.org/msg962288.html) to kernel:)

Reply:
hw counters won't be available if hypervisors don't pass through CPU features to VMs.